cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java

```diff
@@ -162,6 +181,68 @@ public void addTemplatePluginJson(Id.ApplicationTemplate templateId, String file
   }
 
   @Override
+  public void addArtifact(Id.Artifact artifactId, File artifactFile) throws Exception {
+    artifactClient.add(artifactId, null, Files.newInputStreamSupplier(artifactFile));
+  }
+
+  @Override
+  public void addAppArtifact(Id.Artifact artifactId, Class<?> appClass) throws Exception {
+    Location appJar = AppJarHelper.createDeploymentJar(locationFactory, appClass, new Manifest());
+    File destination =
+      new File(tmpFolder, String.format("%s-%s.jar", artifactId.getName(), artifactId.getVersion().getVersion()));
+    Files.copy(Locations.newInputSupplier(appJar), destination);
+    appJar.delete();
+
+    artifactClient.add(artifactId, null, Files.newInputStreamSupplier(destination));
+  }
+
+  @Override
+  public void addPluginArtifact(Id.Artifact artifactId, Id.Artifact parent,
```

Comment:
It might be helpful to add comments for each of these methods to specify what method is used when. It is hard to figure it out from the method arguments.